lib/actions/environment-variables.ts

```diff
@@ -0,0 +1,299 @@
+
+	const response = await balena.request.send({
+		baseUrl: settings.apiUrl,
+		url: '/config/vars',
+	});
+
+	return response.body.reservedNamespaces;
+};
+
+export const list: CommandDefinition<
+	{},
+	{
+		application?: string;
+		device?: string;
+		config: boolean;
+	}
+> = {
+	signature: 'envs',
+	description: 'list all environment variables',
+	help: `\
+Use this command to list all environment variables for
+a particular application or device.
+
+This command lists all application/device environment variables.
+
+If you want to see config variables, used to configure
+balena features, use the --config option.
+
+At the moment the CLI does not support per-service variables,
+so the following commands will only show service-wide
+environment variables.
+
+Example:
+
+	$ balena envs --application MyApp
+	$ balena envs --application MyApp --config
+	$ balena envs --device 7cf02a6\
+`,
+	options: [
+		commandOptions.optionalApplication,
+		commandOptions.optionalDevice,
+
+		{
+			signature: 'config',
+			description: 'show config variables',
+			boolean: true,
+			alias: ['v', 'verbose'],
+		},
+	],
+	permission: 'user',
+	async action(_params, options, done) {
+		normalizeUuidProp(options, 'device');
+		const Bluebird = await import('bluebird');
+		const _ = await import('lodash');
+		const balena = (await import('balena-sdk')).fromSharedOptions();
+		const visuals = await import('resin-cli-visuals');
+
+		const { exitWithExpectedError } = await import('../utils/patterns');
+
+		return Bluebird.try(function(): Promise<
+			DeviceVariable[] | ApplicationVariable[]
+		> {
+			if (options.application != null) {
+				return balena.models.application[
+					options.config ? 'configVar' : 'envVar'
+				].getAllByApplication(options.application);
+			} else if (options.device != null) {
+				return balena.models.device[
+					options.config ? 'configVar' : 'envVar'
+				].getAllByDevice(options.device);
+			} else {
+				return exitWithExpectedError(
+					'You must specify an application or device',
+				);
+			}
+		})
+			.tap(function(environmentVariables) {
+				if (_.isEmpty(environmentVariables)) {
+					exitWithExpectedError('No environment variables found');
+				}
+
+				console.log(
+					visuals.table.horizontal(environmentVariables, [
+						'id',
+						'name',
+						'value',
+					]),
+				);
+			})
+			.nodeify(done);
+	},
+};
+
+export const remove: CommandDefinition<
+	{
+		id: number;
+	},
+	{
+		yes: boolean;
+		device: boolean;
+	}
+> = {
+	signature: 'env rm <id>',
+	description: 'remove an environment variable',
+	help: `\
+Use this command to remove an environment variable from an application.
+
+Notice this command asks for confirmation interactively.
+You can avoid this by passing the \`--yes\` boolean option.
+
+If you want to eliminate a device environment variable, pass the \`--device\` boolean option.
+
+Examples:
+
+	$ balena env rm 215
+	$ balena env rm 215 --yes
+	$ balena env rm 215 --device\
```

Comment:
How about adding a `--config` option as well, so that users can delete config vars (since we provide a way to create config vars).

Reply:
We should, but for now I'd rather burn this to the ground and write it fresh :smiley:. The UX here could be way nicer, pretty easily, the only reason I'm not doing that immediately is that it's a huge breaking change.

For the moment, my goal here is to make sure it doesn't break existing workflows, and then make it powerful and nice to use in a later major bump.

Previously I think this command did actually work for config vars, but nobody appears to be using that (based on the last couple of months of mixpanel data), and the docs had a big warning telling you not to, so I think breaking that is ok.